### PR TITLE
Swift Package Manager support

### PR DESCRIPTION
**Pull request by gfontenot**
This adds support for Swift Package Manager. Only for the production code,
unfortunately. I was looking forward to adding support for `swift test`, but
since we rely on SwiftCheck (which doesn't support Swift Package Manager), we
can't add that support just yet.

### CODE REVIEW

**Comment by RLovelett** [comment]
It would seem that SwiftCheck does support SwiftPM. Or is there something more necessary for support?

https://github.com/typelift/SwiftCheck/commit/7fd76e23ded137e0020a39a854a704b89b1e766d

File: .gitignore [hunk 1]
```diff
@@ -30,6 +30,9 @@ Pods
 # Carthage
 Carthage/
 
+# Swift Package Manager
+.build/
+
```
Comment (RLovelett) [inline .gitignore:35]: Since this is going to depend on SwiftCheck its going to have a Packages directory. Should that be ignored as well?

**Comment by gfontenot** [comment]
tbh, this is _super_ low priority for me right now since nothing about SPM is at all stable. I'll probably revisit this entirely once it's actually released, but till then I'm not going to put too much effort into it.

**Comment by gfontenot** [comment]
Closing this, I'm taking another fresh stab at it.

**Pull request closed without merging**